src/widgets/allergies/allergies-detailed-summary.component.tsx

```diff
@@ -1,120 +1,110 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { Link, useRouteMatch } from "react-router-dom";
-import { performPatientAllergySearch } from "./allergy-intolerance.resource";
-import { createErrorHandler } from "@openmrs/esm-error-handling";
-import styles from "./allergies-detailed-summary.css";
-import SummaryCard from "../../ui-components/cards/summary-card.component";
 import dayjs from "dayjs";
 import { useCurrentPatient } from "@openmrs/esm-api";
-import AllergyForm from "./allergy-form.component";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
 import { openWorkspaceTab } from "../shared-utils";
+import EmptyState from "../../ui-components/empty-state/empty-state.component";
+import SummaryCard from "../../ui-components/cards/summary-card.component";
+import {
+  performPatientAllergySearch,
+  Allergy
+} from "./allergy-intolerance.resource";
+import AllergyForm from "./allergy-form.component";
+import styles from "./allergies-detailed-summary.css";
 
 export default function AllergiesDetailedSummary(
   props: AllergiesDetailedSummaryProps
 ) {
-  const [patientAllergy, setPatientAllergy] = React.useState(null);
-  const [
-    isLoadingPatient,
-    patient,
-    patientUuid,
-    patientErr
-  ] = useCurrentPatient();
+  const [patientAllergies, setPatientAllergies] = useState<Allergy[]>(null);
+  const [isLoadingPatient, patient] = useCurrentPatient();
   const match = useRouteMatch();
 
-  React.useEffect(() => {
+  useEffect(() => {
     if (!isLoadingPatient && patient) {
-      const abortController = new AbortController();
+      const sub = performPatientAllergySearch(
+        patient.identifier[0].value
+      ).subscribe(allergies => {
+        setPatientAllergies(allergies), createErrorHandler();
```

Comment:
This looks equivalent to 

```javascript
      ).subscribe(allergies => {
        setPatientAllergies(allergies);
        createErrorHandler();
      });
```

which does nothing with the error handler created by `createErrorHandler`. Presumably this was not your intention.

Reply:
This is the only thing that absolutely must be fixed.